eve_sqlalchemy/utils.py

```diff
@@ -62,17 +62,20 @@ def sqla_object_to_dict(obj, fields):
 
             # is this field another SQLalchemy object, or a list of SQLalchemy objects?
             if isinstance(val.__class__, DeclarativeMeta):
-                if field in config.DOMAIN:
-                    # we have embedded document in schema, let's resolve it:
-                    result[field] = sqla_object_to_dict(val, list(config.DOMAIN[field]['schema'].keys()))
-                else:
+                try:
+                    foreign_resource = config.DOMAIN[resource]['schema'][field]['data_relation']['resource']
+                    foreign_fields = list(config.DOMAIN[foreign_resource]['schema'].keys())
+                except KeyError:
                     result[field] = getattr(val, config.ID_FIELD)
+                else:
+                    # we have embedded document in schema, let's resolve it:
+                    result[field] = sqla_object_to_dict(val, foreign_fields, foreign_resource)
```

Comment:
I think this will fail since `foreign_resource` may not be set when the `KeyError` exception occurs.